src/NUnitFramework/tests/Constraints/AnyOfConstraintTests.cs

```diff
@@ -0,0 +1,57 @@
+// ***********************************************************************
+// Copyright (c) 2018 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using NUnit.Framework.Internal;
+namespace NUnit.Framework.Constraints
+{
+    [TestFixture]
+    public class AnyOfConstraintTests : ConstraintTestBase
+    {
+        [SetUp]
+        public void SetUp()
+        {
+            theConstraint = new AnyOfConstraint(new[] { 1, 2, 3 });
+            expectedDescription = "any of < 1, 2, 3 >";
+            stringRepresentation = "<anyof 1, 2, 3>";
+        }
+
+        private static object[] SuccessData = new object[] { 1, 2, 3 };
+        private static object[] FailureData = new object[] { new object[] { 4, "4" }, new object[] { "A", "\"A\"" } };
```

Comment:
Just a side comment... not for this PR... Somebody might go through all these and use a `tuple` for the failure data some time. It would be lots clearer to read then.

Reply:
> Somebody might go through all these and use a tuple for the failure data some time. It would be lots clearer to read then.

So we can use C# 7 finally? 😍

Reply:
@jnm2 I thought you guys already decided that. Don't look to me for an OK! 😕

Reply:
Or me :(

Reply:
The last discussion I remember was whether to allow C# 6. I imagine that means a lot of work for Neil, doesn't it?

Reply:
I have a Roslyn program which converts C# 6 to C# 3 (since virtually all of the C# 6 features map to the older CLR).  C# 7 introduces a number of new features which cannot be mapped in that way (such as ref variables).

I would not strongly object to C# 6, although I personally find most of its syntactic sugaring to be undesirable, and in fact we do not allow a lot of it in our coding standards because it makes the code less maintainable.

Reply:
@oznetmaster We currently use C# 6. Too bad there's no way to compile using Roslyn and map the IL instead of mapping syntax?